lib/Cake/Utility/Security.php

```diff
@@ -404,14 +426,42 @@ public static function decrypt($cipher, $key, $hmacSalt = null) {
 			return false;
 		}
 
-		$algorithm = MCRYPT_RIJNDAEL_128;
-		$mode = MCRYPT_MODE_CBC;
-		$ivSize = mcrypt_get_iv_size($algorithm, $mode);
+		if (static::engine() === 'openssl') {
+			$method = 'AES-256-CBC';
+			$ivSize = openssl_cipher_iv_length($method);
+			$iv = substr($cipher, 0, $ivSize);
+			$cipher = substr($cipher, $ivSize);
+			// Regenerate PKCS#7 padding block
+			$padding = openssl_encrypt('', $method, $key, true, substr($cipher, -$ivSize));
+			$plain = openssl_decrypt($cipher . $padding, $method, $key, true, $iv);
+		} else {
+			$algorithm = MCRYPT_RIJNDAEL_128;
+			$mode = MCRYPT_MODE_CBC;
+			$ivSize = mcrypt_get_iv_size($algorithm, $mode);
+			$iv = substr($cipher, 0, $ivSize);
+			$cipher = substr($cipher, $ivSize);
+			$plain = mcrypt_decrypt($algorithm, $key, $cipher, $mode, $iv);
+		}
 
-		$iv = substr($cipher, 0, $ivSize);
-		$cipher = substr($cipher, $ivSize);
-		$plain = mcrypt_decrypt($algorithm, $key, $cipher, $mode, $iv);
 		return rtrim($plain, "\0");
 	}
 
+/**
+ * Set or get the encryption engine
+ *
+ * @param string $engine The encryption engine to use
+ * @return string
+ */
+	public static function engine($engine = null) {
+		if (func_num_args() > 0) {
+			static::$_engine = $engine;
```

Comment:
Should we validate that the chosen engine exists? Right now someone could do `Security::engine('lol')` and not get an error.

Reply:
"lol" powered crypto :muscle:

Reply:
I was lazy a little bit, as I added the method for testing.